test/signalrclienttests/hub_connection_tests.cpp

```diff
@@ -1814,3 +1814,110 @@ TEST(send, throws_if_protocol_fails)
 
     ASSERT_EQ(connection_state::connected, hub_connection->get_connection_state());
 }
+
+TEST(keepalive, sends_ping_messages)
```

Comment:
Should have a test that a ping isn't sent if other messages are being sent from the hub_connection.